jruby/neo4j/driver/ext/exception_mapper.rb

```diff
@@ -2,7 +2,6 @@
 
 java_import org.neo4j.driver.v1.exceptions.AuthenticationException
 java_import org.neo4j.driver.v1.exceptions.ClientException
-java_import org.neo4j.driver.v1.exceptions.NoSuchRecordException
```

Comment:
@lkalwa sorry if we already talked about it but why is this exception being removed?

Reply:
@klobuczek this part is misleading. In exception mapper you're initializing ruby error classes for previously caught `Java::OrgNeo4jDriverV1Exceptions::Neo4jException`, but `NoSuchRecordException` is different type.
See: https://neo4j.com/docs/api/java-driver/current/org/neo4j/driver/v1/exceptions/Neo4jException.html
and https://neo4j.com/docs/api/java-driver/current/org/neo4j/driver/v1/exceptions/NoSuchRecordException.html.
Not sure if this is the right thing to do, but at least we have less unreachable code.

Reply:
@lkalwa I see, this requires some better handling then. There are few more exceptions we need to handle. I think we need to catch all RuntimeExceptions and `e.reraise` should map the known once and re-raise the others unchanged.

Reply:
@klobuczek I suggest to move this part to another PR, as title of this PR is confusing.